packages/engine-test-utils/src/__tests__/engine-server/markdown/noteRefv2.spec.ts

```diff
@@ -684,11 +742,836 @@ describe("noteRefV2", () => {
+        const { engine, vaults } = opts;
+        return processTextV2({
+          text: "# Foo Bar\n![[foo.ch1#^block-anchor]]",
+          dest: opts.extra.dest,
+          engine,
+          vault: vaults[0],
+          fname: "foo",
+        });
+      },
+      preSetupHook: async (opts) => {
+        await ENGINE_HOOKS.setupBasic(opts);
+        await modifyNote(opts, "foo.ch1", (note: NoteProps) => {
+          const txt = [
+            "Laborum libero quia ducimus.",
+            "",
+            "| Sapiente | accusamus |",
+            "|----------|-----------|",
+            "| Laborum  | libero    |",
+            "| Ullam    | optio     |",
+            "^block-anchor",
+            "",
+            "Iure neque alias dolorem.",
+          ];
+          note.body = txt.join("\n");
+          return note;
+        });
+      },
+      verifyFuncDict: {
+        [DendronASTDest.HTML]: async ({ extra }) => {
+          const { resp } = extra;
+          await checkVFile(
+            resp,
+            "Sapiente",
+            "accusamus",
+            "Laborum",
+            "libero",
+            "Ullam",
+            "optio"
+          );
+          await checkNotInVFile(
+            resp,
+            "Minima expedita vero.",
+            "Iure neque alias dolorem."
+          );
+        },
+      },
+    });
+
+    /** When the anchor is anywhere in or after a table, it references the whole table. */
+    const AFTER_TABLE_BLOCK = createProcTests({
+      name: "after table block",
+      setupFunc: async (opts) => {
+        const { engine, vaults } = opts;
+        return processTextV2({
+          text: "# Foo Bar\n![[foo.ch1#^block-anchor]]",
+          dest: opts.extra.dest,
+          engine,
+          vault: vaults[0],
+          fname: "foo",
+        });
+      },
+      preSetupHook: async (opts) => {
+        await ENGINE_HOOKS.setupBasic(opts);
+        await modifyNote(opts, "foo.ch1", (note: NoteProps) => {
+          const txt = [
+            "Laborum libero quia ducimus.",
+            "",
+            "| Sapiente | accusamus |",
+            "|----------|-----------|",
+            "| Laborum  | libero    |",
+            "| Ullam    | optio     |",
+            "",
+            "^block-anchor",
+            "",
+            "Iure neque alias dolorem.",
+          ];
+          note.body = txt.join("\n");
+          return note;
+        });
+      },
+      verifyFuncDict: {
+        [DendronASTDest.HTML]: async ({ extra }) => {
+          const { resp } = extra;
+          await checkVFile(
+            resp,
+            "Sapiente",
+            "accusamus",
+            "Laborum",
+            "libero",
+            "Ullam",
+            "optio"
+          );
+          await checkNotInVFile(
+            resp,
+            "Minima expedita vero.",
+            "Iure neque alias dolorem."
+          );
+        },
+      },
+    });
+
+    runAllTests({
+      name: "compile",
+      testCases: [
```

Comment:
can you also add a test case for what happens when we have a nested block reference link?

Reply:
also a case when a block reference link is referencing to an invalid anchor

Reply:
> can you also add a test case for what happens when we have a nested block reference link?

Do you mean have a note with a reference to another note, that also has a reference? E.g. parsed note has `![[note1#^start:#^end]]`, then note1 has `![[note2#^start:#^end]]`?